docker/build-tools/go.mod

```diff
@@ -0,0 +1,110 @@
+module istio.io/tools/build
+
```

Comment:
is the idea here go.mod would be centralized in the tools repo in the docker container itself?

Reply:
Not quite.

This go.mod file is used to prepropulate the Go module cache within the container. This is suppose to then make it so use of the Go compiler within  the container will not need to redownload these. However, if a go.mod file in a particular repo pulls in more stuff or different versions of the same stuff, it'll just go right into the cache too. But it'll be ephemeral and need to be redownloaded everytime the container is used.

Sadly, this approach hasn't been working. No matter what I load in the container when it is built, all the same modules are redownloaded when I compile something through the container. So somehow, the compiler is refusing to reuse the cached modules. Not sure why yet.

Reply:
is it supposed to be kept in sync with the root `go.mod`?

Reply:
What do you mean by "it" and "root"?

Reply:
Yeah, mostly. We should keep the image up to date with latest versions of things, such that  there is less/no download of modules when compiling repos.

Reply:
instead of a long list of dependencies that may not have all dependencies used by projects would it be possible to just have all the Istio repos listed here? Not sure if that would work but if the stars align that might make it download all the dependencies for all the repos?

Reply:
Yes, I started down that path. It ends up putting our entire source base in the container, so I decided to hold on that.

Before going that route, I want to try getting the module and build cache from the host  rather than the container. That would slow down CI builds since they would need to download all the time, but it would dramatically shrink the container and would speed up iterative developer builds. And it would avoid trying to keep this stuff up to date. This seems like an OK tradeoff.

Reply:
Yeah that makes sense, I think using the host cache is a much better idea. 

The issues with it not using the cache might be related to `module istio.io/tools/build` not matching the module you are building maybe? Just a guess though, not a go modules expert at all

Reply:
Now using the host go caches.